src/pybind/mgr/rook/module.py

```diff
@@ -230,19 +231,7 @@ def get_inventory(self, host_filter=None, refresh=False):
 
         result = []
         for host_name, host_devs in devs.items():
-            devs = []
-            for d in host_devs:
-                dev = inventory.Device(
-                    path='/dev/' + d['name'],
-                    sys_api=dict(
-                        rotational='1' if d['rotational'] else '0',
-                        size=d['size']
-                    ),
-                    available=d['empty'],
-                    rejected_reasons=[] if d['empty'] else ['not empty'],
-                )
-                devs.append(dev)
-
+            devs = [inventory.Device.from_json(json.loads(d['cephVolumeData'])) for d in host_devs]
```

Comment:
Is there a possibility that `cephVolumeData` is not there? eg for public clouds?

Reply:
No. It is obtained from the config map with the information of the devices obtained by the discovery daemon running in each of the k8s nodes.It depends of "ceph-volume" to return data or not for the devices in the node, but the "cephVolumeData" wil be always there ( in the worst case with an empty string)

Reply:
Hey ... I double check the possibility of not having cephVolumeData in the dict .... and [it can happen](https://github.com/rook/rook/blob/ed37a6a699aa70fbc721da7d94b56c7ab79f4609/pkg/util/sys/device.go#L109
)!!
Now i have take into account this and the possibility of being empty.  

Thanks for raise this!